common/common.go

```diff
@@ -241,6 +248,11 @@ const (
 	DefaultGitRetryFactor                    = int64(2)
 )
 
+var (
+	// PersistResourceStatusHealth returns if storing the health check result in the resource status is enabled or not
+	PersistResourceStatusHealth = env.ParseBoolFromEnv(EnvPersistResourceStatusHealth, true)
```

Comment:
Instead of a one-off env var, can this be an argocd-cmd-params-cm item, shared by both the API server and the controller?

Reply:
Agree. Added env variable to `argocd-cmd-params-cm` . At very least it serves as a documentation for new env variable.